nautobot/circuits/urls.py

```diff
@@ -1,150 +1,56 @@
-from django.urls import path
+from django.urls import path, include
 
 from nautobot.dcim.views import CableCreateView, PathTraceView
 from nautobot.extras.views import ObjectChangeLogView
 from . import views
 from .models import Circuit, CircuitTermination, CircuitType, Provider, ProviderNetwork
 
 app_name = "circuits"
+circuit_type_router = views.CircuitTypeRouter()
+provider_router = views.ProviderRouter()
+provider_network_router = views.ProviderNetworkRouter()
+circuit_termination_router = views.CircuitTerminationRouter()
+circuit_router = views.CircuitRouter()
+circuit_type_router.register("circuit-types", views.CircuitTypeViewSet, basename="circuittype")
+provider_router.register("providers", views.ProviderViewSet, basename="provider")
+provider_network_router.register("provider-networks", views.ProviderNetworkViewSet, basename="providernetwork")
+circuit_termination_router.register(
+    "circuit-terminations", views.CircuitTerminationViewset, basename="circuittermination"
+)
+circuit_router.register("circuits", views.CircuitViewSet, basename="circuit")
```

Comment:
The reasoning I think is that we have to specify the `basename=` to make `name= ` for each model's set of views ( it was originally in `path()`: cirucuittermination_edit, cirucuittermination_delete) as well as `prefix` "circuits", "circuit-terminations" etc for the actual url. and the `viewset= `argument is mandatory.
<img width="1445" alt="Screen Shot 2022-07-06 at 10 12 26 AM" src="https://user-images.githubusercontent.com/46973263/177570826-bb3247bd
So for circuit detail view the reverse url for `circuittermination_edit` will not exist because we failed to register the `basename `for it in the router. 
<img width="1605" alt="Screen Shot 2022-07-06 at 10 27 07 AM" src="https://user-images.githubusercontent.com/46973263/177573940-23be7586-746c-4b1d-882e-71906d564cc4.png">
-bf7f-4ee3-b0af-d1126e79eb60.png">
if we did `circuit_termination_router.register("circuits", views.CircuitViewSet, basename="circuittermination")`. The edit, delete button will render but it shows the error above.

Reply:
> I'm a bit confused as to why we have both a custom Router and a custom ViewSet for each model, and why we have to tell the Router which ViewSet to use. Seems error-prone - what happens if I do `circuit_termination_router.register("circuits", views.CircuitViewSet, basename="circuit")` by mistake?

I have similar concerns @glennmatthews which I addressed in a suggestion here: https://github.com/nautobot/nautobot/pull/1902#discussion_r915174666

I understand why we have a ViewSet and a Router (a Route points to a ViewSet). My suggestion attempts to make this coupling tighter, and I am unaware what it would take to collapse this further...

It seems all we would need in this file would be for `NautobotViewSet` (and the ViewSets inheriting from it) supply a `.urls` attribute? I don't know all of what `SimpleRouter` does vs the collection of inheritance on `NautobotViewSet` as it stands.